### ISSUE TITLE
[Bug] Incorrect type declaration for async functions for transifex/api

### ISSUE THREAD

**Issue by gera2ld** [transifex/transifex-javascript#153]
**Current Behavior**

https://github.com/transifex/transifex-javascript/blob/fff2621eb1b451d61ba3ffab3febb0d85fdc50c5/packages/api/src/transifexApi.d.ts#L44-L47

![image](https://user-images.githubusercontent.com/3139113/194768749-723d266f-6935-444c-8546-0307a469f231.png)

**Expected Behavior**

The type declaration should be:

```ts
declare class Collection { 
   async fetch(): Promise<void>; 
   async getNext(): Promise<Collection>;
   async getPrevious(): Promise<Collection>;
// etc.
```

**Comment by nbasili** [comment]
@gera2ld I think [this PR](https://github.com/transifex/transifex-javascript/pull/155) will fix both what you described and some other instances. What do you think?

**Comment by nbasili** [comment]
Related fixes are part of [4.3.0](https://github.com/transifex/transifex-javascript/releases/tag/v4.3.0) release. Thanks for reporting this 🙏
